picard/ui/options/plugins_download.py

```diff
@@ -0,0 +1,204 @@
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+
+import os
+import sys
+import zipfile
+from functools import partial
+from PyQt4 import QtCore, QtGui
+from picard import config, log
+from picard.const import (USER_DIR, USER_PLUGIN_DIR,
+                          USER_DOWNLOADS_DIR, PICARD_URLS, PLUGINS_API)
+from picard.util import encode_filename, webbrowser2
+from picard.webservice import XmlWebService
+from picard.ui.options import OptionsPage, register_options_page
+from picard.ui.ui_options_plugins_download import Ui_PluginsDownloadPage
+
+
+class PluginsDownloadPage(OptionsPage):
+
+    NAME = "plugins_download"
+    TITLE = N_("Download")
+    PARENT = "plugins"
+    SORT_ORDER = 70
+    ACTIVE = True
+
+    options = [
+        config.ListOption("setting", "enabled_plugins", []),
+    ]
+
+    def __init__(self, parent=None):
+        super(PluginsDownloadPage, self).__init__(parent)
+        self.ui = Ui_PluginsDownloadPage()
+        self.ui.setupUi(self)
+        self.items = {}
+        self.ui.plugins.itemSelectionChanged.connect(self.change_details)
+        self.ui.plugins.setSortingEnabled(True)
+        self.ui.plugins.mimeTypes = self.mimeTypes
+        if sys.platform == "win32":
+            self.loader = "file:///%s"
+        else:
+            self.loader = "file://%s"
+        self.ui.install_plugin.clicked.connect(self.download_plugin)
+        self.ui.install_plugin.setEnabled(False)
+        self.ui.open_repo.clicked.connect(self.open_repo)
+        self.tagger.pluginmanager.plugin_installed.connect(self.plugin_installed)
+        self.installed = [p.module_name for p in self.tagger.pluginmanager.plugins]
+        self.plugins_available = self.tagger.plugins_available
+
+    def load(self):
+        self.ui.plugins.clear()
+        self.ui.details.setText("<b>"+ _("No new plugins available.") + "</b>")
+        for module_name, data in self.plugins_available.items():
+            data['module_name'] = module_name
+            if module_name not in self.installed:
+                item = self.add_plugin_item(data)
+        self.ui.plugins.setCurrentItem(self.ui.plugins.topLevelItem(0))
+
+    def add_plugin_item(self, plugin, enabled=False, item=None):
+        if item is None:
+            item = QtGui.QTreeWidgetItem(self.ui.plugins)
+        item.setText(0, plugin['name'])
+        item.setText(1, plugin['version'])
+        item.setText(2, str(plugin['downloads']))
```

Comment:
Yes, it was removed because of the server-side implementation, which was modifying the json file to store download count, leading to unnecessary complications (handling updates etc...) 
Please just drop related code for now, we'll eventually re-introduce this feature later (and it has to use separated data storage).